oak_crypto/src/noise_nk/mod.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This was copied from Chromium's third_party/cloud_authenticator, which has
+//! compatible copyright and ownership (Apache 2.0, Google).
+
+#[allow(unused_imports)] // Macros only used in tests.
+#[macro_use]
+
+mod error;
+mod noise;
+mod rustcrypto;
+
+use alloc::vec::Vec;
+use core::result::Result;
+
+pub use crate::noise_nk::rustcrypto::{
+    aes_256_gcm_open_in_place, aes_256_gcm_seal_in_place, ecdsa_verify, hkdf_sha256,
+    p256_scalar_mult, rand_bytes, sha256, sha256_two_part, EcdsaKeyPair, P256Scalar,
+};
+use crate::noise_nk::{
+    error::Error,
+    noise::{HandshakeType, Noise, NONCE_LEN},
+};
+
+// This is assumed to be vastly larger than any connection will ever reach.
+const MAX_SEQUENCE: u32 = 1u32 << 24;
+
+/// The length of an uncompressed, X9.62 encoding of a P-256 point.
+pub const P256_X962_LENGTH: usize = 65;
+
+#[derive(Debug)]
```

Comment:
I think this may be suspicious for external reviewers, since we "may" log secret keys